### PR TITLE
Lucia codesniffer meqp1

### PR DESCRIPTION
**Pull request by luciaebizmarts**
cs #779 Compliance for ECG and MEQP coding standards

### CODE REVIEW

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php [hunk 1]
```diff
@@ -213,15 +214,15 @@ protected function GeneratePOSTPayload($order, $mailchimpStoreId, $magentoStoreI
...
+            $productSyncData = $helper->getEcommerceSyncDataItem($productId, $this->isProduct(), $mailchimpStoreId);
             if ($item->getProductType() == Mage_Catalog_Model_Product_Type::TYPE_CONFIGURABLE) {
                 $options = $item->getProductOptions();
                 $sku = $options['simple_sku'];
                 $variant = Mage::getModel('catalog/product')->getIdBySku($sku);
                 if (!$variant) {
                     continue;
                 }
-            } elseif ($item->getProductType() == Mage_Catalog_Model_Product_Type::TYPE_BUNDLE || $item->getProductType() == Mage_Catalog_Model_Product_Type::TYPE_GROUPED) {
+            } elseif ($item->getProductType() == $this->typBundle() || $item->getProductType() == $this->typGrouped()) {
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php]: Fix typo in typBundle and typGrouped.

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php [hunk 2]
```diff
@@ -715,12 +721,84 @@ protected function makeSalesRule()
...
+    /**
+     * @return string
+     */
+    protected function typGrouped()
+    {
+        return Mage_Catalog_Model_Product_Type::TYPE_GROUPED;
+    }
+
+    /**
+     * @return false|Mage_Core_Model_Abstract
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php]: return type: Ebizmarts_MailChimp_Model_Api_Customers

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php [hunk 3]
```diff
@@ -715,12 +721,84 @@ protected function makeSalesRule()
...
+     */
+    protected function makeApiCustomers()
+    {
+        return Mage::getModel('mailchimp/api_customers');
+    }
+
+    /**
+     * @return false|Mage_Core_Model_Abstract
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php]: Return type: Mage_Directory_Model_Country

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php [hunk 4]
```diff
@@ -715,12 +721,84 @@ protected function makeSalesRule()
...
+    /**
+     * @return string
+     */
+    protected function migrateLastOrderId()
+    {
+        return Ebizmarts_MailChimp_Model_Config::GENERAL_MIGRATE_LAST_ORDER_ID;
+    }
+
+    /**
+     * @return false|Mage_Core_Model_Abstract
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php]: Return type: Ebizmarts_MailChimp_Model_Api_Products

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Stores.php [hunk 5]
```diff
@@ -38,7 +38,7 @@ public function createMailChimpStore($mailChimpStoreId, $listId = null, $scopeId
                 $storeDomain = $helper->getStoreDomain($scopeId, $scope);
                 if (strpos($storeEmail, 'example.com') !== false) {
                     $storeEmail = null;
-                    throw new Exception('Please, change the general email in Store Email Addresses/General Contact');
+                    Mage::throwException('Please, change the general email in Store Email Addresses/General Contact');
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Stores.php]: Add translation as follows: $helper->__('Text');

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Stores.php [hunk 6]
```diff
@@ -55,13 +55,19 @@ public function createMailChimpStore($mailChimpStoreId, $listId = null, $scopeId
...
+                    $strB = ($strC . ' Possible solutions ');
+                    $strA = 'A MailChimp store with the same domain already exists in this account.';
+                    $errorMessage = $helper->__($strA . $strB . $str);
                     $adminSession->addError($errorMessage);
                 }
                 $helper->logError($e->getFriendlyMessage());
             }
         } else {
-            throw new Exception('You don\'t have any lists configured in MailChimp');
+            Mage::throwException('You don\'t have any lists configured in MailChimp');
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Stores.php]: Add translation as follows: $helper->__('Text');

File: app/code/community/Ebizmarts/MailChimp/Model/Email/Template.php [hunk 7]
```diff
@@ -191,8 +190,9 @@ public function getMail()
             return $this->_mail;
         } else {
             $storeId = Mage::app()->getStore()->getId();
-            Mage::helper('mailchimp/mandrill')->log("store: $storeId API: " . Mage::getStoreConfig(Ebizmarts_MailChimp_Model_Config::MANDRILL_APIKEY, $storeId), $storeId);
-            $this->_mail = new Mandrill_Message(Mage::getStoreConfig(Ebizmarts_MailChimp_Model_Config::MANDRILL_APIKEY, $storeId));
+            $storeConfig = Mage::getStoreConfig(Ebizmarts_MailChimp_Model_Config::MANDRILL_APIKEY, $storeId);
+            $this->makeMandrillHelper()->log("store: $storeId API: " . $storeConfig, $storeId);
```
Comment (Santiagoebizmarts) [inline app/code/community/Ebizmarts/MailChimp/Model/Email/Template.php]: Add translation as follows: $helper->__('Text');

**Review by SantiParagarino — CHANGES_REQUESTED** [review]

**Review by centerax — CHANGES_REQUESTED** [review]

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Products.php [hunk 8]
```diff
@@ -352,35 +361,38 @@ protected function _updateSyncData($productId, $mailchimpStoreId, $syncDelta = n
         );
     }
 
-    public function buildProductDataRemoval($product, $batchId, $mailchimpStoreId)
-    {
-        //@Todo handle product removal
-        return array();
+/*
```
Comment (centerax) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Products.php]: delete commented code.

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php [hunk 9]
```diff
@@ -577,29 +620,30 @@ public function replaceAllOrdersBatch($initialTime, $mailchimpStoreId, $magentoS
...
-            $config = array(array(Ebizmarts_MailChimp_Model_Config::GENERAL_MIGRATE_LAST_ORDER_ID, $orderId));
+            $config = array(array($this->migrateLastOrderId(), $orderId));
             if (!$helper->timePassed($initialTime)) {
-                $batchArray[$this->_counter]['method'] = "DELETE";
+                $batchArray[$this->_counter]['method'] = 'DELETE';
                 $batchArray[$this->_counter]['path'] = '/ecommerce/stores/' . $mailchimpStoreId . '/orders/' . $orderId;
                 $batchArray[$this->_counter]['operation_id'] = $this->_batchId . '_' . $orderId;
                 $batchArray[$this->_counter]['body'] = '';
-                $this->_counter += 1;
+                ++$this->_counter;
```
Comment (centerax) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php]: leave as it was

File: app/code/community/Ebizmarts/MailChimp/Model/Api/PromoCodes.php [hunk 10]
```diff
@@ -158,7 +198,16 @@ public function makePromoCodesCollection($magentoStoreId)
     protected function makeDeletedPromoCodesCollection($mailchimpStoreId)
     {
         $deletedPromoCodes = Mage::getModel('mailchimp/ecommercesyncdata')->getCollection();
-        $deletedPromoCodes->getSelect()->where("mailchimp_store_id = '" . $mailchimpStoreId . "' AND type = '" . Ebizmarts_MailChimp_Model_Config::IS_PROMO_CODE . "' AND mailchimp_sync_deleted = 1");
+        $maChimpSyncDel = "' AND mailchimp_sync_deleted = 1";
```
Comment (centerax) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/PromoCodes.php]: verify that strings are same

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Stores.php [hunk 11]
```diff
@@ -38,30 +38,50 @@ public function createMailChimpStore($mailChimpStoreId, $listId = null, $scopeId
...
+                );
                 return $response;
 
             } catch (Ebizmarts_MailChimp_Helper_Data_ApiKeyException $e) {
                 $helper->logError($e->getMessage());
             } catch (MailChimp_Error $e) {
                 $adminSession = Mage::getSingleton('adminhtml/session');
                 if (strstr($e->getFriendlyMessage(), 'A store with the domain')) {
-                    $errorMessage = $helper->__('A MailChimp store with the same domain already exists in this account. You need to have a different URLs for each scope you set up the ecommerce data. Possible solutions ') . "<a href='https://docs.magento.com/m1/ce/user_guide/search_seo/seo-url-rewrite-configure.html'>HERE</a> and <a href='https://docs.magento.com/m1/ce/user_guide/configuration/url-secure-unsecure.html'>HERE</a>";
+                    $urlA = "'https://docs.magento.com/m1/ce/user_guide/search_seo/seo-url-rewrite-configure.html'";
```
Comment (centerax) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Stores.php]: use " or ', only one.

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Stores.php [hunk 12]
```diff
@@ -38,30 +38,50 @@ public function createMailChimpStore($mailChimpStoreId, $listId = null, $scopeId
...
+                    $urlB = "'https://docs.magento.com/m1/ce/user_guide/configuration/url-secure-unsecure.html'";
+                    $str = "<a href=". $urlA .">HERE</a> "."and <a href=". $urlB .">HERE</a>";
+                    $strC = ' You need to have a different URLs for each scope you set up the ecommerce data.';
+                    $strB = ($strC . ' Possible solutions ');
```
Comment (centerax) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Stores.php]: remove `(`

File: app/code/community/Ebizmarts/MailChimp/Model/Api/Stores.php [hunk 13]
```diff
@@ -38,30 +38,50 @@ public function createMailChimpStore($mailChimpStoreId, $listId = null, $scopeId
...
+                    $strA = 'A MailChimp store with the same domain already exists in this account.';
+                    $errorMessage = $helper->__($strA . $strB . $str);
```
Comment (centerax) [inline app/code/community/Ebizmarts/MailChimp/Model/Api/Stores.php]: check string is same

File: app/code/community/Ebizmarts/MailChimp/Model/Email/Queue.php [hunk 14]
```diff
@@ -89,7 +89,7 @@ public function send()
                         }
 
                         unset($mailer);
-                        $message->setProcessedAt(Varien_Date::formatDate(true));
+                        $message->setProcessedAt(Mage::getSingleton('core/date')->gmtDate());
```
Comment (centerax) [inline app/code/community/Ebizmarts/MailChimp/Model/Email/Queue.php]: check that are same and extract

**Review by centerax — CHANGES_REQUESTED** [review]

**Review by centerax — APPROVED** [review]

**Pull request closed without merging**